### PR TITLE
quiz complete

### PR DESCRIPTION
**Pull request by ddayporter**
Comfort: 5
Completeness: 5

I keep mixing up how to access keys in a hash (vs js objects), but I think I got it right.. Also, not sure if you guys meant to leave the comments in there, but I got a laugh out of them so thanks. :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,12 +54,14 @@ add her to the list of guests in the castle.
 
 Write your code here:
 ```ruby
-# code here
+
+ town[:castle][:guests] << town[:residents].delete_at(1)
```
Comment (mattscilipoti) [inline README.md:58]: :+1:

File: README.md [hunk 2]
```diff
@@ -71,7 +78,11 @@ Belle is friends with Mrs. Potts
 
 Write your code here:
 ```ruby
-# code here
+
+friends.each do |friend|
+  puts "Belle is friends with #{friend}"
+end
```
Comment (mattscilipoti) [inline README.md:84]: :+1:

File: README.md [hunk 3]
```diff
@@ -81,7 +92,9 @@ Write your code here:
 Describe what an ERD is, and why we create them for applications. Also give an
 example what the attributes and relationships might be for the following
 entities (no need to draw an ERD):
-<!-- Maybe clarify whether they're meant to give relationships between all four entities or... -->
+<!-- Maybe clarify whether they're meant to give relationships between all four entities or...
+I see this comment and taking it to mean we DON'T need to assign relationships
+between all 4-->
```
Comment (mattscilipoti) [inline README.md:97]: Sounds like a reasonable translation to me.

File: README.md [hunk 4]
```diff
@@ -89,7 +102,23 @@ entities (no need to draw an ERD):
...
+skin color, type of wrist manacles, and foreign keys for Lamp ID and Master ID.
+Lamp's attributes might include metal type, number of handles, lid shape, and
+foreign key for Genie ID.
+
+Person and Pet would have a many-to-many relationship because a person could have
+many pets, and a pet could belong to many people (each member of a family, for
+example). A person's attributes might include name, gender, age, marital status,
+job, hobbies, and birthday (depending on the purpose of the app, this list could
+go on or be much shorter). A pet's attributes might include name, gender, age,
+breed, temperament and size.
```
Comment (mattscilipoti) [inline README.md:121]: :tada:  Beautiful.  Any relationship between a Person and a Genie?  :)

File: README.md [hunk 5]
```diff
@@ -100,7 +129,11 @@ SQL database. If you need an example, you can use: people and wishes
 
 Your answer:
 ```
-Replace this with your answer
+A schema lays out the structure of the tables within the database, and sets the
+data type for each column (name would be a VARCHAR type, age would be INTEGER).
+A one-to-many relationship is represented by adding the foreign key of the "one"
+as an attribute on the "many". So, each wish would have a "person_id" attribute
+that would refer back to the person.
```
Comment (mattscilipoti) [inline README.md:136]: :+1:  Couldn't have said it better myself.  There has to be a "The needs of the many outweigh the needs of the one." pun in there somewhere.  There has to be.

File: README.md [hunk 6]
```diff
@@ -125,7 +158,17 @@ Write code to do the following:
...
+
+lamp = Lamp.create(wishes_remaining: 3)
+lamp.genies.create(name: 'Genie')
+lamp[:wishes_remaining] = 1
+lamp.save
+
+new_lamp = Lamp.create(wishes_remaining: 3)
+new_lamp.genies.create(name: 'Jafar')
+
+Genie.find_by(name: 'Genie').update(lamp_id: nil)
```
Comment (mattscilipoti) [inline README.md:170]: :+1:  You're on fire today.

File: README.md [hunk 7]
```diff
@@ -140,15 +183,47 @@ would look like for such an application.
...
+  * This is the new route, which displays a form where you can enter information
+  about a new warrior.
+
 GET '/warriors/:id'
   * This is the show route, which finds a warrior by ID, and displays information about that warrior.
 
+GET '/warriors/:id/edit'
+  * This is the edit route, which finds a warrior by ID and displays a page that
+  includes a delete button, should your warrior fall in battle, or just displeases you,
+  and a form where you can update a specific warrior's information.
```
Comment (mattscilipoti) [inline README.md:206]: Those disappointing warriors, never quite giving that 110% that we have grown to expect.  At least they're easy to replace. <click>.

File: README.md [hunk 8]
```diff
@@ -140,15 +183,47 @@ would look like for such an application.
...
+  * This is the update route, assuming the form on the '/warriors/:id/edit' page
+  had an input element of type hidden, method 'put', action '/warriors'. It should
+  include a method updating a warrior using params from the form, and redirect
+  to '/warriors'.
+
+DELETE '/warriors'
+  * This is the destroy route, assuming the delete button on the edit page had an
+  input type of hidden, method 'delete', action '/warriors'. It should include a
+  method to delete the warrior referenced in the params[:id] and redirect to
+  '/warriors'.
```
Comment (mattscilipoti) [inline README.md:224]: :aries:  
/me Raising shield and sword to show respect for those who have fought bravely and well (you) and those who have fallen (no names).

File: README.md [hunk 9]
```diff
@@ -170,5 +245,12 @@ Write what an example ERB file (aka view) might look like to list all the warrio
...
 Write your code here (**NOTE: syntax highlighting doesn't work for ERB in markdown files, so ignore the colors!**):
 ```html
-<!-- code here -->
+
+<h2>Warriors</h2>
+<ul>
+<% @warriors.each do |warrior| %>
+  <li><%= warrior[:name] %></li> 
+<% end %>
+</ul>
```
Comment (mattscilipoti) [inline README.md:254]: :+1:

**Comment by mattscilipoti** [comment]
You nailed it.  I saw no evidence of confusion.  Thanks for making it fun.

**Comment by ddayporter** [comment]
I don't know if replying to this in my email will work but - thanks Matt! I
felt really good about it.

**Pull request closed without merging**